src/esp/gfx/GenericDrawable.cpp

```diff
@@ -34,47 +36,61 @@ GenericDrawable::GenericDrawable(
       shaderManager_{shaderManager},
       lightSetup_{shaderManager.get<LightSetup>(lightSetupKey)},
       materialData_{
-          shaderManager.get<MaterialData, PhongMaterialData>(materialDataKey)},
+          shaderManager.get<Mn::Trade::MaterialData,
+                            Mn::Trade::PhongMaterialData>(materialDataKey)},
       skinData_(skinData),
       jointTransformations_() {
   flags_ = Mn::Shaders::PhongGL::Flag::ObjectId;
 
   /* If texture transformation is specified, enable it only if the material is
      actually textured -- it's an error otherwise */
-  if (materialData_->textureMatrix != Mn::Matrix3{} &&
-      (materialData_->ambientTexture || materialData_->diffuseTexture ||
-       materialData_->specularTexture || materialData_->specularTexture ||
-       materialData_->textureObjectId)) {
+  if (materialData_->commonTextureMatrix() != Mn::Matrix3{} &&
+      (materialData_->hasAttribute("ambientTexturePtr") ||
+       materialData_->hasAttribute("baseColorTexturePtr") ||
+       materialData_->hasAttribute("diffuseTexturePtr") ||
+       materialData_->hasAttribute("specularTexturePtr") ||
+       materialData_->attribute<bool>("hasTextureObjectId"))) {
```

Comment:
Continuing from what I said above,

```suggestion
       materialData_->hasAttribute("textureObjectId"))) {
```

instead of a second boolean attribute.